Sources/Datadog/FeaturesIntegration/WebView/WebRUMEventConsumer.swift

```diff
@@ -29,32 +29,92 @@ internal class WebRUMEventConsumer: WebEventConsumer {
         switch eventType {
         case "view":
             let viewEvent = try jsonDecoder.decode(RUMViewEvent.self, from: eventData)
-            let mappedViewEvent = mapIfNeeded(dataModel: viewEvent, context: rumContext, offset: getOffset(viewID: viewEvent.view.id))
+            let mappedViewEvent = mapIfNeeded_RUMViewEvent(dataModel: viewEvent, context: rumContext, offset: getOffset(viewID: viewEvent.view.id))
             write(mappedViewEvent)
         case "action":
             let actionEvent = try jsonDecoder.decode(RUMActionEvent.self, from: eventData)
-            let mappedActionEvent = mapIfNeeded(dataModel: actionEvent, context: rumContext, offset: getOffset(viewID: actionEvent.view.id))
+            let mappedActionEvent = mapIfNeeded_RUMActionEvent(dataModel: actionEvent, context: rumContext, offset: getOffset(viewID: actionEvent.view.id))
             write(mappedActionEvent)
         case "resource":
             let resourceEvent = try jsonDecoder.decode(RUMResourceEvent.self, from: eventData)
-            let mappedResourceEvent = mapIfNeeded(dataModel: resourceEvent, context: rumContext, offset: getOffset(viewID: resourceEvent.view.id))
+            let mappedResourceEvent = mapIfNeeded_RUMResourceEvent(dataModel: resourceEvent, context: rumContext, offset: getOffset(viewID: resourceEvent.view.id))
             write(mappedResourceEvent)
         case "error":
             let errorEvent = try jsonDecoder.decode(RUMErrorEvent.self, from: eventData)
-            let mappedErrorEvent = mapIfNeeded(dataModel: errorEvent, context: rumContext, offset: getOffset(viewID: errorEvent.view.id))
+            let mappedErrorEvent = mapIfNeeded_RUMErrorEvent(dataModel: errorEvent, context: rumContext, offset: getOffset(viewID: errorEvent.view.id))
             write(mappedErrorEvent)
         case "long_task":
             let longTaskEvent = try jsonDecoder.decode(RUMLongTaskEvent.self, from: eventData)
-            let mappedLongTaskEvent = mapIfNeeded(dataModel: longTaskEvent, context: rumContext, offset: getOffset(viewID: longTaskEvent.view.id))
+            let mappedLongTaskEvent = mapIfNeeded_RUMLongTaskEvent(dataModel: longTaskEvent, context: rumContext, offset: getOffset(viewID: longTaskEvent.view.id))
             write(mappedLongTaskEvent)
         default:
             userLogger.error("🔥 Web RUM Event Error - Unknown event type: \(eventType)")
         }
     }
 
-    private func mapIfNeeded<T: RUMDataModel>(dataModel: T, context: RUMContext?, offset: Offset) -> T {
-        // TODO: RUMM-1786 implement mutating session_id & application_id
-        return dataModel
+    private func mapIfNeeded_RUMViewEvent(dataModel: RUMViewEvent, context: RUMContext?, offset: Offset) -> RUMViewEvent {
+        guard let context = context,
+              context.sessionID != .nullUUID else {
```

Comment:
when the session is sampled-out, its `sessionID` is set to `.nullUUID`.
with this condition, we send web events with their browser SDK context if the native session is sampled-out.